java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -183,7 +194,7 @@ public AccessToken refreshAccessToken() throws IOException {
                 }
             }
 
-            if (isCurrentTokenValid(this.currentToken, clock.instant())) {
+            if (needsRefresh(this.currentToken, clock.instant())) {
```

Comment:
sorry I missed this change, semantics are inverted: if needs refresh then return current token
Method should be negated or kept as `isCurrentTokenValid`